tests/test_bgp_af.rb

```diff
@@ -126,9 +172,13 @@ def test_default_information_originate
     assert(bgp_af.default_information_originate,
            'Error: default-information originate not set')
 
-    pattern = /^ *default-information originate$/
-    af_string = get_bgp_af_cfg(asn, vrf, af)
-
+    if platform == :nexus
+      pattern = /^ *default-information originate$/
+      af_string = get_bgp_af_cfg(asn, vrf, af)
+    elsif platform == :ios_xr
+      pattern = /^ default-information originate$/
+      af_string = @device.cmd("show run router bgp | in default-information")
```

Comment:
> On 20 Nov 2015, at 16:47, Glenn Matthews <notifications@github.com> wrote:
> 
> In tests/test_bgp_af.rb <https://github.com/cisco/cisco-network-node-utils/pull/87#discussion_r45482597>:
> 
> > @@ -126,9 +172,13 @@ def test_default_information_originate
> >      assert(bgp_af.default_information_originate,
> >             'Error: default-information originate not set')
> >  
> > -    pattern = /^ *default-information originate$/
> > -    af_string = get_bgp_af_cfg(asn, vrf, af)
> > -
> > +    if platform == :nexus
> > +      pattern = /^ *default-information originate$/
> > +      af_string = get_bgp_af_cfg(asn, vrf, af)
> > +    elsif platform == :ios_xr
> > +      pattern = /^ default-information originate$/
> > +      af_string = @device.cmd("show run router bgp | in default-information")
> Were you planning to address this and make get_bgp_af_cfg work for XR?
> 
> 
The above was the old code.  The new code is:

 173     elsif platform == :ios_xr
 174       af_string = @device.cmd("show run router bgp #{asn} vrf #{vrf} " \
 175                               'default-information originate’)

get_bp_af_cfg is used for multiple commands on nexus
but there’s no equivalent in iOS_xr that can be used for all
so I use the appropriate extended show run router bgp command in each case.
> —
> Reply to this email directly or view it on GitHub <https://github.com/cisco/cisco-network-node-utils/pull/87/files#r45482597>.
>